src/Robo/Plugin/Commands/CICommands.php

```diff
@@ -145,41 +100,43 @@ public function jobFixCodingStandards(): Result
      */
     protected function jobRunCodingStandards(bool $applyFixes = false): Result
     {
+        $customCodePaths = $this->getCustomCodePaths();
+        $phpcsStandards = implode(',', $this->getRequiredRoboConfigArrayFor('phpcs_standards'));
+        $phpcsCheckExtensions = implode(',', $this->getRequiredRoboConfigArrayFor('phpcs_check_extensions'));
+        $phpcsIgnorePaths = implode(',', $this->getRequiredRoboConfigArrayFor('phpcs_ignore_paths'));
+        $phpcsPhpVersion = Robo::config()->get('phpcs_php_version', $this::PHPCS_DEFAULT_PHP_VERSION);
```

Comment:
I kind of like some of these things that require "processing" i.e. calling `implode` or things that are "global" being set in the constructor to allow re-use of them, but perhaps that is over-optimizing?

Reply:
The issue I ran into is that the constructor is called even when just calling `robo` without a command so that it can discover and list all the commands. This works fine if every single one of the configuration values is set or optional, but if you want to configure one as required and throw an exception if it is not set, you would be throwing an exception in a scenario when the actual command is not even being run.

For that reason, I decided to ditch this approach of configuring things in the constructor _especially_ in classes that expose multiple commands.